pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -431,6 +432,16 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 	} else {
 		xStorageInfo := new(info.StorageInfo)
 		xStorageInfo.StorageType = info.StorageTypeInfo_STORAGE_TYPE_INFO_EXT4
+		mi, err := mount.Lookup(types.PersistDir)
+		if err != nil {
+			log.Errorf("cannot find device with %s mount", types.PersistDir)
```

Comment:
May be instead of logging error, it might be good to send the device not mounted status to UI
rDiskStatus.Status = info.StorageStatus_STORAGE_STATUS_NOT_MOUNTED ?

Reply:
Well, right now it is really unexpected situation. And we still have no information about disk name in that case

Reply:
I mean code should be something like this.

                        rDiskStatus := new(info.StorageDiskState)
 			rDiskStatus.DiskName = new(evecommon.DiskDescription)
 			rDiskStatus.DiskName.Name = *proto.String(mi.Source)
                       if err != nil {
                        rDiskStatus.Status = info.StorageStatus_STORAGE_STATUS_NOTMOUNED
                       } else {
 			rDiskStatus.Status = info.StorageStatus_STORAGE_STATUS_ONLINE
                      }
 			xStorageInfo.Disks = append(xStorageInfo.Disks, rDiskStatus)

Reply:
Oh I see you are saying if not mounted we will not get the device name anyway and hence nothing to send. But sending blank disk is OK with status not mounted. That is valid case.

Reply:
We cannot use possibly undefined `mi.Source` in case of error

Reply:
Spoke with @giggsoff offline and since if persist is not even mounted device will be in suspect state anyway and cannot even send this state. 

This commit LGTM.